### ISSUE TITLE
When training the dataset, Nan appears

### ISSUE THREAD

**Issue by zhaijs** [TexasInstruments/edgeai-yolov5#33]
## When I started trying to train the COCO2017 dataset, first it was normal, but then there was Nan, which I thought might be the cause of the exploding gradient, but I didn't make any changes to the code. This question has been bothering me for a long time and I look forward to your suggestions.
![~%BTTIS7CBF REIC_SK$F@2](https://user-images.githubusercontent.com/42409347/176828994-0610171b-3c0b-4f23-a607-f060d600fd57.png)
![38A49470C2786F8A255E172DBDCE40CD_750_750](https://user-images.githubusercontent.com/42409347/176829083-f085860b-f4a7-45ae-bcbb-4758e7ca27f7.jpg)

**Comment by zhaijs** [comment]
There may be something wrong with OKS loss and I'm trying to use a different loss function

**Comment by zhaijs** [comment]
It looks like the number is too small to cause the nan to appear, less than 0.01

**Comment by wanglegoc** [comment]
hi @zhaijs  Do you solved this problem, would be nice to share some solutions?
Thank you

**Comment by GeorgeInbelgium** [comment]
I meet the same problem, can you share some solutions？

**Comment by GeorgeInbelgium** [comment]
I find some of my label files are empty. my solution is to modify the label files.

**Comment by zhaijs** [comment]
I checked the dataset and found no errors.
When I calculated kpt_loss_factor, the denominator added 1e-9.
`kpt_loss_factor = (torch.sum(kpt_mask != 0) + torch.sum(kpt_mask == 0))/(torch.sum(kpt_mask != 0)+1e-9)`

**Comment by liamsun2019** [comment]
I met the same issue. Any solution or workaround for it?

**Comment by gh244** [comment]
This happens when I first set batchsize to 1,You can try making the batchsize bigger
